master/buildbot/test/unit/test_scripts_base.py

```diff
@@ -320,11 +321,61 @@ def test_checkBasedir_no_dir(self):
     @skipUnlessPlatformIs('posix')
     def test_checkBasedir_active_pidfile(self):
         self.activeBasedir()
-        open(os.path.join('test', 'twistd.pid'), 'w').close()
+        # write our own pid in the file
+        with open(os.path.join('test', 'twistd.pid'), 'w') as f:
+            f.write(str(os.getpid()))
         rv = base.checkBasedir(mkconfig())
         self.assertFalse(rv)
         self.assertInStdout('still running')
 
+    @skipUnlessPlatformIs('posix')
+    def test_checkBasedir_bad_pidfile(self):
+        self.activeBasedir()
+        # write our own pid in the file
```

Comment:
`write our own pid in the file` - should be `write invalid string as pid in the file` or something like that